tests/traffic/test_api_perf.py

```diff
@@ -0,0 +1,244 @@
+import pytest
+import utils
+import traceback
+import time
+from abstract_open_traffic_generator import flow, control, result
+
+
+CSV_COLUMNS = [
+    'config_type', 'iterations', 'set_config_ms', 'start_flows_ms',
+    'stop_flows_ms', 'start_capture_ms', 'stop_capture_ms',
+    'get_port_results_ms', 'get_flow_results_ms', 'get_captures_ms', 'result'
+]
+CSV_DIR = utils.new_logs_dir()
+
+
+@pytest.mark.e2e
+@pytest.mark.parametrize(
+    'config_type', ['one_flow_tcp_128b', 'ten_flows_tcp_128b']
+)
+@pytest.mark.parametrize("iterations", [10])
+def test_api_perf(settings, config_type, iterations):
+    """
+    This test collects avg duration in milliseconds for all required API calls
+    required in a usual E2E test script.
+    """
+    packets = 100
+    row = new_result_dict(config_type, iterations)
+
+    for i in range(1, iterations + 1):
```

Comment:
as "iterations" is parametrize variable, instead of using for loop can leverage the iteration mechanism of pytest.

@pytest.mark.parametrize("iterations", list(range(10)))

Reply:
I agree, the only problem is that we'd have to think about managing snippets living outside the loop. e.g L57-60. Any suggestions on that ?

Reply:
can be made that as fixture and scope it, but it doesn't make sense just for one script to do so.
if these tests grow can think of it later.